stdlib.sh

```diff
@@ -356,7 +356,7 @@ path_add() {
 
   # prepend the passed paths in the right order
   for ((i = $#; i > 0; i--)); do
-    path_array=("$(expand_path "${!i}")" "${path_array[@]}")
+    path_array=("$(expand_path "${!i}")" "${path_array[@]+${path_array[@]}}")
```

Comment:
> There is a big table of compatible versions, with each their own errors:
> https://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u/61551944#61551944

JFYI: following the link there's an explicit note re the syntax:
```
Quoting the full expansion instead of the nested array ("${arr[@]+${arr[@]}}"), which I would have expected to be roughly equivalent, is similarly unsafe in 4.2.
```

Reply:
In https://github.com/direnv/direnv/pull/667 I had this:
```suggestion
    path_array=("$(expand_path "${!i}")" ${path_array[@]+"${path_array[@]}"})
```
before dropping that commit.

Reply:
oops, good catch